### DISCUSSION TITLE
Bagaimana cara membuat json data recent post by vote:user

### DISCUSSION THREAD

**Discussion by yuliakbard** [mecha-cms/mecha#293] [Q&A]
kalau menggunakan cara langsung untuk menampilkan recent post by vote:user, ane liat, utk loading muat halaman pertama kali itu, ada cukup banyak jeda, rencananya mau ane konversi dalam bentuk json dulu, baru ditampilin ke halaman menggunakan javascript.

Apa ada feed.json yg menampilkan recent post by vote:user kang? atau bisakah dibuat data json nya?

**Comment by taufik-nurrohman** [comment]
Sebenarnya nggak semua harus dijadikan JSON dan dimuat dengan JavaScript. Cache bisa juga dibuat dalam bentuk berkas. Misalnya seperti ini:

``` php
$cache = LOT . D . 'cache' . D . 'recent-votes.php';

$pages = [];

if (exist($cache, 1)) {
    // Load cache
    $pages = require $cache;
} else {
    // Create cache
    foreach ($user->getVotesAsPages() as $page) {
        $pages[] = $page->path;
    }
    // Save cache to a file
    save($cache, '<?' . 'php return' . z($pages) . ';', 0600);
}

$pages = new Pages($pages);
```

**Reply by yuliakbard** [reply]
`$cache = LOT . D . 'cache' . D . 'recent-votes.php'; // ini isinya apa kang? recent-votes.php`

**Reply by yuliakbard** [reply]
ane dah liat dalamnya kang, ane paham

**Reply by yuliakbard** [reply]
di dalam folder cache, kalau ada folder lagi, kehitung cache juga ya kang?

`$cache = LOT . D . 'cache' . D . 'posts' . D . $page->name . '.php';`

**Reply by yuliakbard** [reply]
cache ini mau ane coba manfaatin, recent post by filter username juga nih kang, jadi kalo cache itu ane pisahin jadi `cache.php` atau jadi sebuah fungsi yg bisa dipakai berulang kali, apa bisa kah kang?

**Reply by yuliakbard** [reply]
ada pertanyaan lagi nih kang, setelah ane cobain, saat ada penambahan votes sebuah posting, jumlah yg ditampilkan hanya menyesuaikan dengan jumlah yg ada di cache file? kapan terjadi pembaruan cache nya kang? atau hanya segitu saja kah?

**Reply by taufik-nurrohman** [reply]
> di dalam folder cache, kalau ada folder lagi, kehitung cache juga ya kang?
> 
> `$cache = LOT . D . 'cache' . D . 'posts' . D . $page->name . '.php';`

Iya. Nama cache mah bebas&hellip;

**Reply by taufik-nurrohman** [reply]
> cache ini mau ane coba manfaatin, recent post by filter username juga nih kang, jadi kalo cache itu ane pisahin jadi `cache.php` atau jadi sebuah fungsi yg bisa dipakai berulang kali, apa bisa kah kang?

Ya paling masing-masing simpan sebagai nama user:

``` php
$cache = LOT . D . 'cache' . D . 'recent-votes' . D . $user->name . '.php';
```

**Reply by taufik-nurrohman** [reply]
> ada pertanyaan lagi nih kang, setelah ane cobain, saat ada penambahan votes sebuah posting, jumlah yg ditampilkan hanya menyesuaikan dengan jumlah yg ada di cache file? kapan terjadi pembaruan cache nya kang? atau hanya segitu saja kah?
> 

Cache tidak berubah/diperbarui kecuali dihapus. Bisa si dibuat aksi menghapus setiap kali _vote_ dilakukan:

``` php
Hook::set(['on.vote.let', 'on.vote.set'], function () use ($cache) {
    delete($cache);
});
```

**Reply by yuliakbard** [reply]
![image](https://github.com/user-attachments/assets/7c55cc3e-0fd8-4766-8f4f-bed6e451af5f)
cem mana nih kang? 😅

**Reply by taufik-nurrohman** [reply]
> ![image](https://github.com/user-attachments/assets/7c55cc3e-0fd8-4766-8f4f-bed6e451af5f)
> cem mana nih kang? 😅

Kurang `]` setelah `'on.vote.set'` hehehehe.

**Reply by yuliakbard** [reply]
kang, ini `Hook::set() ` ini taruh di index.php atau di page.php aja kang?, soalnya gk ada perubahan

**Reply by taufik-nurrohman** [reply]
Dalem `index.php`. Kalau nggak ngaruh coba argumen `$stack` ditambah.
